modules/lfs/pointer.go

```diff
@@ -21,32 +26,80 @@ const (
 	MetaFileOidPrefix = "oid sha256:"
 )
 
-// TryReadPointer tries to read LFS pointer data from the reader
-func TryReadPointer(reader io.Reader) *Pointer {
+var (
+	// ErrMissingPrefix occurs if the content lacks the LFS prefix
+	ErrMissingPrefix = errors.New("Content lacks the LFS prefix")
+
+	// ErrInvalidStructure occurs if the content has an invalid structure
+	ErrInvalidStructure = errors.New("Content has an invalid structure")
+
+	// ErrInvalidOIDLength occurs if the oid has an invalid length
+	ErrInvalidOIDLength = errors.New("OID has an invalid length")
+)
+
+// ReadPointer tries to read LFS pointer data from the reader
+func ReadPointer(reader io.Reader) (Pointer, error) {
 	buf := make([]byte, blobSizeCutoff)
-	n, _ := io.ReadFull(reader, buf)
+	n, err := io.ReadFull(reader, buf)
```

Comment:
```suggestion
	// io.ReadFull only ever returns io.EOF if 0 bytes are read.
	n, err := io.ReadFull(reader, buf)
```

Might be helpful to handle io.EOF on line 45 and convert to a more understandable error.

Reply:
An error in `ReadPointer` or `ReadPointerFromBuffer` just indicates that the content is not a pointer. The error isn't of real interest here. Maybe I could change the method signatures to return `(p Pointer, ok bool, err error)`. Then `ok` would be checked and not `if err == nil`. What do you think?

Reply:
no worries.